apps/src/gamelab/GameLabP5.js

```diff
@@ -581,6 +582,33 @@ GameLabP5.prototype.afterSetupComplete = function() {
   this.p5._setupEpiloguePhase2();
 };
 
+GameLabP5.prototype.preloadBackgrounds = function() {
+  if (this.preloadedBackgrounds) {
+    this.p5._predefinedBackgrounds = this.preloadedBackgrounds;
+    return Promise.resolve();
+  }
+  this.preloadedBackgrounds = {};
+  this.p5._predefinedBackgrounds = {};
+  return Promise.all(
+    backgrounds.map(background => {
+      return new Promise(resolve => {
+        this.p5.loadImage(
+          background.sourceUrl,
+          image => {
+            this.preloadedBackgrounds[background.legacyParam] = image;
+            this.p5._predefinedBackgrounds[background.legacyParam] = image;
+            resolve();
+          },
+          err => {
+            console.log(err);
+            resolve();
```

Comment:
Should this be `reject`?

Reply:
What should happen if a background fails to load? With this change, it would be a silent failure, which you would only notice if you actually try to use that background (it just wouldn't show)
I feel like we wouldn't want to block you from doing anything, especially because we're preloading for everyone, and presumably a lot of projects don't use the background images at all.. Maybe the error dialog?